test/functional/api/vim_spec.lua

```diff
@@ -1037,4 +1037,31 @@ describe('api', function()
     end)
   end)
 
+  describe('nvim_get_keyboard_mods', function()
+    it('returs a dictionary on or the specific error', function()
+      -- Returns the dictionary on success otherwise a specified error
+      -- if the resources were not present on the system.
+
+      local status, rv = pcall(meths.get_keyboard_mods)
+      if status then
+        for k, v in pairs(rv) do
+          eq(true, ({CapsLock=true, NumLock=true, ScrollLock=true})[k])
+          eq('boolean', type(v))
+        end
+      else
+        if os_name() == 'unix' then
+          local errMch = string.match(rv, "Unable to load *")
```

Comment:
`<space>*` means “space repeated zero or more times” and not “space followed by any number of any characters”. I.e. having asterisk here is absolutely pointless, removing it will not change conditions under which tests pass at all, though theoretically may alter the error message (i.e. make “*error message part* then two spaces” fail with “*error message part* then space” not equal to “*error message part* then two spaces” error rather then “*error message part* then space” not equal to `nil`).

Reply:
And, BTW, `string` “module” needs not be used, `string.match(str, re)` is the same thing as `str:match(re)` as long as `str` has string type.